### ISSUE TITLE
False positive: tabindex rule

### ISSUE THREAD

**Issue by dylanb** [dequelabs/axe-core#251]
The following HTML is generating a violation with the tabindex rule and should not

```
<!DOCTYPE html>
<html lang="en">
<head>
<title>Testing tabindex of -1 on elements inside a focusable element.</title>
</head>
<body>
<h1>Examples.</h1>

<h2>A span with tabindex="-1" in a link</h2>
<a href="http://www.google.com"><span tabindex="-1">Google</span></a>
<h2>SVG element with tabindex="-1" in a link</h2>
<a href="http://www.foursquare.com">
<svg version="1.1" width="300" height="200" role="presentation" focusable="false" tabindex="-1">
<title></title>
<rect width="75" height="50" rx="20" ry="20" fill="#90ee90" stroke="#228b22" stroke-fill="1" />
</svg>
Foursquare
</a>
<h2>SVG with tabindex="-1" not in a link</h2>
<svg version="1.1" width="300" height="200" role="presentation" focusable="false" tabindex="-1">
<title></title>
<rect width="75" height="50" rx="20" ry="20" fill="#90ee90" stroke="#228b22" stroke-fill="1" />
</svg>
</body>
</html>
```

**Comment by sarbbottam** [comment]
I could not reproduce it with [axe-core@2.0.7 - latest published version](https://www.npmjs.com/package/axe-core)

```diff
<!DOCTYPE html>
<html lang="en">
<head>
<title>Testing tabindex of -1 on elements inside a focusable element.</title>
</head>
<body>
<h1>Examples.</h1>

<h2>A span with tabindex="-1" in a link</h2>
<a href="http://www.google.com"><span tabindex="-1">Google</span></a>
<h2>SVG element with tabindex="-1" in a link</h2>
<a href="http://www.foursquare.com">
<svg version="1.1" width="300" height="200" role="presentation" focusable="false" tabindex="-1">
<title></title>
<rect width="75" height="50" rx="20" ry="20" fill="#90ee90" stroke="#228b22" stroke-fill="1" />
</svg>
Foursquare
</a>
<h2>SVG with tabindex="-1" not in a link</h2>
<svg version="1.1" width="300" height="200" role="presentation" focusable="false" tabindex="-1">
<title></title>
<rect width="75" height="50" rx="20" ry="20" fill="#90ee90" stroke="#228b22" stroke-fill="1" />
</svg>
</body>
+<script src="node_modules/axe-core/axe.min.js" ></script>
+<script>
+  axe.a11yCheck(document, function (results) {
+    console.log(results. violations); // [] - empty array
+});
+</script>
</html>
```

**Comment by dylanb** [comment]
@sarbbottam you are correct!

**Comment by Wildebrew** [comment]
I have tested this bug using v1.02 of aXe for Firefox and Chrome, and this is still throwing a positive tbindex value for examples 2 and 3 (as well as throwing over 20 violations on a website weh I am using aXe).
This was last tested on Firefox 49 running on Window 8.1.

Could this be a problem with the browser plug-in interaction, or is it possible that I am not getting the latest version of aXe?

**Comment by dylanb** [comment]
@Wildebrew the extension executes the best practice rules which will complain about positive index values. In the API, you can turn these off by telling it to execute only wcag2a and wcag2aa rules.

**Comment by Wildebrew** [comment]
Thanks
That helps a bit (though when you use the browser extensions you can't turn off the best practice rules, or can you?)
This rule is still falsely reporting use of positive tabindex values whenever tabindex="-1" is used on an <svg> element (examples 2 and 3 in the uploaded source code).
I would recommend the rule be modified to not report use of tabindex="-1" on an <svg> element, the description of the error does not apply to this case.
A separate rule could be written to test for presence of tabindex values on <svg> elements, though I would just recommend not checking <svg> under this rule (a separate rule or set of rules can be written to test <svg> images.

**Comment by dylanb** [comment]
@marcysutton can you test what @Wildebrew is saying and validate its fixed in the new extensions we have submitted?

@Wildebrew you can only turn rules off and on currently in FireEyes II :-(

**Comment by marcysutton** [comment]
I am getting 0 violations with those from the code on master. The extensions are still in the process of being updated, so it will be fixed when those are released.

**Comment by Wildebrew** [comment]
If this bug is gone in the next version of the aXe plug-ins I will be happy.
&lt;svg&gt; is a relatively new frontier for accessibility checks, so understandably it takes some trials and tribulations to get it going.

**Comment by marcysutton** [comment]
The new axe-chrome extension is out, and I verified there are no issues with the above HTML file.
